### ISSUE TITLE
Trying to add maximum flexibility and put things in order in this mess too

### ISSUE THREAD

**Issue by Alexustas** [Alexustas/Props-requests#48]
I spent the last couple of weeks on a complete redesign and refinement of such an important prop as a toggle switch

short [demo-video](https://youtu.be/FFBknDTboHM)

"ModularToggleSwitch_v0.2" [complete guide](
https://www.dropbox.com/s/vmnwpnh9pbu4xlh/ModularToggleSwitch_v0.2.pdf?dl=0)

So, what has been done:

- Altered or modified models of the base and levers type 1 and 2
- Added levers of types 3, 4 and 5 with different color variations of the caps
- The proportions of the guards (type 1 and 2) were changed, now they are wider, as in a real spacecraft
- New types of guards were added
- New types of backlight were added
- Protective covers type 1 and 2 were added
- User's guide was completely redesigned and expanded

Latest dev-version of ASET stuff
DL: https://www.dropbox.com/s/0375isnayjvqazo/ASET_Props-Dev1.4.2-31-03-2017.zip?dl=0

I plan to gradually replace all available toggle switches with new ones and in the future completely remove the old version from the package

**Comment by MOARdV** [comment]
That is excellent.  Modular switch design will definitely allow a lot of customization.

**Comment by DeputyLOL** [comment]
Thanks for this

**Comment by theonegalen** [comment]
Oh wow. Time to redo all my switches, I guess! :D

**Comment by Alexustas** [comment]
New Push-Button prop

VIDEO: https://youtu.be/cjjfMRuF3Ec

![bandicam 2017-05-03 14-17-03-685](https://cloud.githubusercontent.com/assets/6713671/25654619/38a78148-300b-11e7-929a-4d41eff5c450.jpg)
![bandicam 2017-05-03 14-16-55-811](https://cloud.githubusercontent.com/assets/6713671/25654618/38a3eea2-300b-11e7-9fcd-e0389cdf8e75.jpg)
![bandicam 2017-05-03 14-16-49-883](https://cloud.githubusercontent.com/assets/6713671/25654617/38a3067c-300b-11e7-9119-9cbfd512c8e2.jpg)
![bandicam 2017-05-03 14-16-31-341](https://cloud.githubusercontent.com/assets/6713671/25654620/38a7ff7e-300b-11e7-844c-97e1c1491bf2.jpg)
![bandicam 2017-05-03 14-18-09-694](https://cloud.githubusercontent.com/assets/6713671/25654666/6067c5e4-300b-11e7-8785-92aaa5041859.jpg)

**Comment by Alexustas** [comment]
"Modular Push-button" User Guide (WIP)
https://www.dropbox.com/s/ob0qlb9ofe35ptu/ModularPushButton_V0.2.pdf?dl=0

**Comment by Alexustas** [comment]
Latest "ASET Props" Dev-build
DL: https://www.dropbox.com/s/8mb1ho9ygep5uvx/ASET_Props-Dev%201.4.3-03-05-2017.zip?dl=0

**Comment by MOARdV** [comment]
I am making some props using these.  The documentation is very good.  One thing I do not see - where are the transforms for labels?  If I look at your example props, I see TextTopObj in some props, but the Modular Toggle Switch v0.2 document does not identify the label transforms.

EDIT: Also, the backlight model base numbers in Module Toggle Switch is off by one.  In the document, the page shows Bcklt_10 as the first model (two horizontal lines backlight), but that is Bcklt_11 when I create a prop.

**Comment by Alexustas** [comment]
>  but the Modular Toggle Switch v0.2 document does not identify the label transforms.

Yes you are right. I will try to add a description in the near future

**Comment by Alexustas** [comment]
quick illustration

![bandicam 2017-05-11 01-12-43-220](https://cloud.githubusercontent.com/assets/6713671/25919085/f7c3ab02-35e6-11e7-8aeb-c3975ca10484.jpg)

**Comment by Alexustas** [comment]
` In the document, the page shows Bcklt_10 as the first model (two horizontal lines backlight), but that is Bcklt_11 when I create a prop.`

Toggle Switch Manual   https://www.dropbox.com/s/vmnwpnh9pbu4xlh/ModularToggleSwitch_v0.2.pdf?dl=0

**Comment by MOARdV** [comment]
Okay, Thank you.

**Comment by MOARdV** [comment]
![pb_transparency](https://cloud.githubusercontent.com/assets/6089493/26114152/2704db04-3a22-11e7-869e-08b32d688ff1.png)

Modular push button cover pb_Cover02 blocks emissive.  Note the emissive green "OK" is visible when the cover is up, but it is invisible when the cover is down.  The "RECV" has a diffuse white.  This is using MAS, not RPM, but I do not think that should make a difference.

**Comment by Alexustas** [comment]
It is very strange. I use this cover for many props and everything works as it should

https://youtu.be/0GstAsGqovg

Do you use uniform scaling? Can i see the full config?

**Comment by MOARdV** [comment]
> Do you use uniform scaling? Can i see the full config?

Yes.  scaling is 0.5, 0.5, 0.5 on that prop.  Configs attached, but it is for MAS, and you will need to update MAS before the prop will work.  I will try to update MAS in the next day.

[MAS_pb.zip](https://github.com/Alexustas/Props-requests/files/1005102/MAS_pb.zip)

EDIT: I created another prop and placed it below this prop, and it works, just like the props in your video.  It must be something unusual about the camera position.

EDIT 2: I checked the visibility from the left-hand seat, and the emissive text is not visible on either prop.  I think it may be related to viewing angle (maybe z-fighting?).  I will do some more experiments to see if I can understand what is happening.

**Comment by Alexustas** [comment]
Do you still have this problem? I slightly tested your internals and could not reproduce this issue

**Comment by MOARdV** [comment]
> Do you still have this problem? I slightly tested your internals and could not reproduce this issue

I have not used the cover for a while.  I will add it onto some props and test again later today.

**Comment by MOARdV** [comment]
New question: for model `ASET/ASET_Props/Control/Switch_Toggle_Modular/models/TgglLever_Type_2`, KSP hangs when I include the `texture =` substitution in the prop:
```
MODEL
{
	model = ASET/ASET_Props/Control/Switch_Toggle_Modular/models/TgglLever_Type_2
	texture = Switch_TUMBLEDiffuse,ASET/ASET_Props/Control/Switch_Tumble/Switch_TUMBLEDiffuse
}
```
If I do not use `texture =`, it is okay.

And the `SwitchGrp` transform does not appear to be correct (using a rotation on the x axis, the lever moves left-right at an angle).  Is that the incorrect transform?

**Comment by MOARdV** [comment]
> Do you still have this problem? I slightly tested your internals and could not reproduce this issue

I can not reproduce it now, either.
